internal/controller/nopresource/nopresource.go

```diff
@@ -217,7 +189,29 @@ func (c *external) Delete(ctx context.Context, mg resource.Managed) error {
 
 	fmt.Printf("Deleting: %+v", cr)
 
-	cr.Status.SetConditions(runtimev1alpha1.Deleting())
+	cr.Status.SetConditions(xpv1.Deleting())
 
 	return nil
 }
+
+func ReconcileLogic(conditionAfter []v1alpha1.ResourceConditionAfter, t time.Duration) []int {
```

Comment:
because this function is UpperCase, it will be an exported function.  There's no reason to export this function since it's only called within this package, so you can call it `reconcileLogic` instead to keep it from being exported.